### PR TITLE
DAOS-4463 csum: Checksum Fault Injection Cleanup

### PR DESCRIPTION
**Pull request by ryon-jensen**
- Remove checksum fault injection from VOS. Doesn't really belong
   because there is no checksum calculations in the VOS layer.
 - Make fault names more consistent and meaningful.
 - Don't need separate client/server faults. Can just set appropriately
   using daos_fail_loc_set(client) vs daos_mgmt_set_params(server)

Signed-off-by: Ryon Jensen <ryon.jensen@intel.com>

### CODE REVIEW

File: src/object/srv_obj.c [hunk 1]
```diff
@@ -975,6 +975,14 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 	if (obj_rpc_is_update(rpc)) {
 		obj_singv_ec_rw_filter(orw, iods, offs, true);
 		bulk_op = CRT_BULK_GET;
+
+		/** Fault injection - corrupt data from network */
+		if (DAOS_FAIL_CHECK(DAOS_CSUM_CORRUPT_UPDATE)  && !rma) {
```
Comment (liuxuezhao) [inline src/object/srv_obj.c:980]: it is only to corrupt data for !rma (RPC inline)? actually bulk data also can be corrupt if you want, crt_bulk_access() can get the sgl and then corrupt it.

File: src/object/srv_obj.c [hunk 2]
```diff
@@ -1063,8 +1071,19 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
...
 
-	rc = obj_verify_bio_csum(rpc, iods, iod_csums, biod,
-				 cont_hdl->sch_csummer);
+	if (obj_rpc_is_update(rpc)) {
+		rc = obj_verify_bio_csum(rpc, iods, iod_csums, biod,
+					 cont_hdl->sch_csummer);
+		/** CSUM Verified on update, now corrupt to fake corruption
+		 * on disk
+		 */
+		if (DAOS_FAIL_CHECK(DAOS_CSUM_CORRUPT_DISK) && !rma) {
```
Comment (liuxuezhao) [inline src/object/srv_obj.c:1080]: here the data already written to disk right? just curious how would the below dcf_corrupt help?

File: src/object/srv_obj.c (hunk 1 above)
Comment (ryon-jensen) [reply src/object/srv_obj.c:980]: True. Bulk data will get corrupted in obj_bulk_transfer

File: src/object/srv_obj.c (hunk 2 above)
Comment (ryon-jensen) [reply src/object/srv_obj.c:1080]: I was expecting that data hasn't been written here. (isn't that what bio_iod_post does?)

**Comment by ryon-jensen** [comment]
All failures are known:
- Bandit
- Pool evict test (DAOS-4410)

**Pull request merged**